### ISSUE TITLE
Testsuite fails

### ISSUE THREAD

**Issue by SirVer** [cournape/Bento#35]
I was unable to find how to properly run the test suite, so I just executed 
$ nosetests 

after python setup.py install in the root directory.

Version is: 5efb0e8c70 (BUG: fix _compute_extension_name when local_node is the top node.).

Some tests fail, this is the complete output:

```
FF.............................................................................FFFFFFFFFFFFFFFF.FF........E.......
======================================================================
ERROR: test_compiled_library (test_subpackage.TestTranslation)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/test_subpackage.py", line 83, in test_compiled_library
    spkg, top_node)
  File "/Users/sirver/Desktop/Promotion/prog/Bento/bento/core/subpackage.py", line 92, in flatten_subpackage_compiled_libraries
    sources = resolve_glob(clib.sources, spkg.rdir)
  File "/Users/sirver/Desktop/Promotion/prog/Bento/bento/core/utils.py", line 232, in resolve_glob
    ret.extend(expand_glob(f, reference_path))
  File "/Users/sirver/Desktop/Promotion/prog/Bento/bento/core/utils.py", line 83, in expand_glob
    raise IOError("no files following pattern %s found" % pattern)
IOError: no files following pattern src/clib.c found

======================================================================
FAIL: parsing.functionals.test.test_sphinx
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/site-packages/nose/case.py", line 186, in runTest
    self.test(*self.arg)
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/functionals/test.py", line 34, in test_sphinx
    _test_functional("sphinx")
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/functionals/test.py", line 31, in _test_functional
    assert_equal(tested, ref, "divergence for %s" % info)
AssertionError: divergence for /Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/functionals/sphinx.info

======================================================================
FAIL: parsing.functionals.test.test_jinja2
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/site-packages/nose/case.py", line 186, in runTest
    self.test(*self.arg)
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/functionals/test.py", line 37, in test_jinja2
    _test_functional("jinja2")
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/functionals/test.py", line 31, in _test_functional
    assert_equal(tested, ref, "divergence for %s" % info)
AssertionError: divergence for /Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/functionals/jinja2.info

======================================================================
FAIL: test_flag (parsing.test_visitor.TestConditional)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 286, in test_flag
    assert_equal(parse_and_analyse(data), descr)
AssertionError: {'flag_options': {'debug': {'default': 'true', 'name': 'debug'}}, 'libraries': {'default': {'name': 'default', 'install_requires': [], 'py_modules': ['foo.py', 'bar.py'], 'build_requires': [], 'extensions': {}, 'packages': [], 'compiled_libraries': {}}}} != {'executables': {}, 'extra_sources': [], 'flag_options': {'debug': {'default': 'true', 'name': 'debug'}}, 'libraries': {'default': {'name': 'default', 'install_requires': [], 'py_modules': ['foo.py', 'bar.py'], 'build_requires': [], 'extensions': {}, 'packages': [], 'compiled_libraries': {}}}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_literal (parsing.test_visitor.TestConditional)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 241, in test_literal
    assert_equal(parse_and_analyse(data), descr)
AssertionError: {'libraries': {'default': {'name': 'default', 'install_requires': [], 'py_modules': ['foo.py', 'bar.py'], 'build_requires': [], 'extensions': {}, 'packages': [], 'compiled_libraries': {}}}} != {'executables': {}, 'extra_sources': [], 'flag_options': {}, 'libraries': {'default': {'name': 'default', 'install_requires': [], 'py_modules': ['foo.py', 'bar.py'], 'build_requires': [], 'extensions': {}, 'packages': [], 'compiled_libraries': {}}}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_os (parsing.test_visitor.TestConditional)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 267, in test_os
    assert_equal(parse_and_analyse(data), descr)
AssertionError: {'libraries': {'default': {'name': 'default', 'install_requires': [], 'py_modules': ['foo.py', 'bar.py'], 'build_requires': [], 'extensions': {}, 'packages': [], 'compiled_libraries': {}}}} != {'executables': {}, 'extra_sources': [], 'flag_options': {}, 'libraries': {'default': {'name': 'default', 'install_requires': [], 'py_modules': ['foo.py', 'bar.py'], 'build_requires': [], 'extensions': {}, 'packages': [], 'compiled_libraries': {}}}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_simple (parsing.test_visitor.TestDataFiles)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 208, in test_simple
    assert_equal(parse_and_analyse(data), descr)
AssertionError: {'data_files': {'man1doc': {'files': ['foo.1'], 'target_dir': '/usr/share/man/man1', 'name': 'man1doc', 'source_dir': 'doc/man1'}}} != {'executables': {}, 'extra_sources': [], 'flag_options': {}, 'libraries': {}, 'data_files': {'man1doc': {'files': ['foo.1'], 'target_dir': '/usr/share/man/man1', 'name': 'man1doc', 'source_dir': 'doc/man1'}}, 'path_options': {}}

======================================================================
FAIL: test_nested_indented_block3 (parsing.test_visitor.TestDescription)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 120, in test_nested_indented_block3
    assert_equal(parse_and_analyse(data), ref)
AssertionError: {'description': 'some\nsimple\n    description\n        on\n\nmultiple\n\nlines.'} != {'executables': {}, 'extra_sources': [], 'description': 'some\nsimple\n    description\n        on\n\nmultiple\n\nlines.', 'flag_options': {}, 'libraries': {}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_simple_indented_block (parsing.test_visitor.TestDescription)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 85, in test_simple_indented_block
    assert_equal(parse_and_analyse(data), ref)
AssertionError: {'description': 'some simple description\non multiple\nlines.'} != {'executables': {}, 'extra_sources': [], 'description': 'some simple description\non multiple\nlines.', 'flag_options': {}, 'libraries': {}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_simple_indented_block2 (parsing.test_visitor.TestDescription)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 96, in test_simple_indented_block2
    assert_equal(parse_and_analyse(data), ref)
AssertionError: {'description': 'some simple description\non multiple\nlines.'} != {'executables': {}, 'extra_sources': [], 'description': 'some simple description\non multiple\nlines.', 'flag_options': {}, 'libraries': {}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_simple_single_line (parsing.test_visitor.TestDescription)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 73, in test_simple_single_line
    assert_equal(parse_and_analyse(data), ref)
AssertionError: {'description': 'some simple description'} != {'executables': {}, 'extra_sources': [], 'description': 'some simple description', 'flag_options': {}, 'libraries': {}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_modules (parsing.test_visitor.TestExecutable)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 318, in test_modules
    assert_equal(parse_and_analyse(data), descr)
AssertionError: {'executables': {'foo': {'function': 'main', 'name': 'foo', 'module': 'foo.bar'}}} != {'executables': {'foo': {'function': 'main', 'name': 'foo', 'module': 'foo.bar'}}, 'extra_sources': [], 'flag_options': {}, 'libraries': {}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_simple (parsing.test_visitor.TestFlag)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 225, in test_simple
    assert_equal(parse_and_analyse(data), descr)
AssertionError: {'flag_options': {'debug': {'default': 'false', 'name': 'debug', 'description': 'debug flag'}}} != {'executables': {}, 'extra_sources': [], 'flag_options': {'debug': {'default': 'false', 'name': 'debug', 'description': 'debug flag'}}, 'libraries': {}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_build_requires (parsing.test_visitor.TestLibrary)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 160, in test_build_requires
    assert_equal(parse_and_analyse(data), descr)
AssertionError: {'libraries': {'default': {'name': 'default', 'install_requires': [], 'py_modules': [], 'build_requires': ['foo'], 'extensions': {}, 'packages': [], 'compiled_libraries': {}}}} != {'executables': {}, 'extra_sources': [], 'flag_options': {}, 'libraries': {'default': {'name': 'default', 'install_requires': [], 'py_modules': [], 'build_requires': ['foo'], 'extensions': {}, 'packages': [], 'compiled_libraries': {}}}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_install_requires (parsing.test_visitor.TestLibrary)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 172, in test_install_requires
    assert_equal(parse_and_analyse(data), descr)
AssertionError: {'libraries': {'default': {'name': 'default', 'install_requires': ['foo'], 'py_modules': [], 'build_requires': [], 'extensions': {}, 'packages': [], 'compiled_libraries': {}}}} != {'executables': {}, 'extra_sources': [], 'flag_options': {}, 'libraries': {'default': {'name': 'default', 'install_requires': ['foo'], 'py_modules': [], 'build_requires': [], 'extensions': {}, 'packages': [], 'compiled_libraries': {}}}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_modules (parsing.test_visitor.TestLibrary)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 134, in test_modules
    assert_equal(parse_and_analyse(data), descr)
AssertionError: {'libraries': {'default': {'name': 'default', 'install_requires': [], 'py_modules': ['foo.py'], 'build_requires': [], 'extensions': {}, 'packages': [], 'compiled_libraries': {}}}} != {'executables': {}, 'extra_sources': [], 'flag_options': {}, 'libraries': {'default': {'name': 'default', 'install_requires': [], 'py_modules': ['foo.py'], 'build_requires': [], 'extensions': {}, 'packages': [], 'compiled_libraries': {}}}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_simple_extension (parsing.test_visitor.TestLibrary)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 148, in test_simple_extension
    assert_equal(parse_and_analyse(data), descr)
AssertionError: {'libraries': {'default': {'name': 'default', 'install_requires': [], 'py_modules': [], 'build_requires': [], 'extensions': {'_foo': {'sources': ['foo.c'], 'name': '_foo', 'include_dirs': []}}, 'packages': [], 'compiled_libraries': {}}}} != {'executables': {}, 'extra_sources': [], 'flag_options': {}, 'libraries': {'default': {'name': 'default', 'install_requires': [], 'py_modules': [], 'build_requires': [], 'extensions': {'_foo': {'sources': ['foo.c'], 'name': '_foo', 'include_dirs': []}}, 'packages': [], 'compiled_libraries': {}}}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_simple (parsing.test_visitor.TestPath)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 189, in test_simple
    assert_equal(parse_and_analyse(data), descr)
AssertionError: {'path_options': {'manpath': {'default': '/usr/share/man', 'name': 'manpath', 'description': 'man path'}}} != {'executables': {}, 'extra_sources': [], 'flag_options': {}, 'libraries': {}, 'data_files': {}, 'path_options': {'manpath': {'default': '/usr/share/man', 'name': 'manpath', 'description': 'man path'}}}

======================================================================
FAIL: test_author (parsing.test_visitor.TestSimpleMeta)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 65, in test_author
    assert_equal(parse_and_analyse(data), ref)
AssertionError: {'author': 'John Doe'} != {'executables': {}, 'extra_sources': [], 'author': 'John Doe', 'flag_options': {}, 'libraries': {}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_name (parsing.test_visitor.TestSimpleMeta)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 47, in test_name
    assert_equal(parse_and_analyse(data), ref)
AssertionError: {'name': 'foo'} != {'executables': {}, 'extra_sources': [], 'name': 'foo', 'flag_options': {}, 'libraries': {}, 'data_files': {}, 'path_options': {}}

======================================================================
FAIL: test_summary (parsing.test_visitor.TestSimpleMeta)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sirver/Desktop/Promotion/prog/Bento/tests/parsing/test_visitor.py", line 56, in test_summary
    assert_equal(parse_and_analyse(data), ref)
AssertionError: {'summary': 'a few words.'} != {'executables': {}, 'extra_sources': [], 'flag_options': {}, 'summary': 'a few words.', 'libraries': {}, 'data_files': {}, 'path_options': {}}

----------------------------------------------------------------------
Ran 114 tests in 0.552s

FAILED (errors=1, failures=20)
```

**Comment by cournape** [comment]
Running nosetests is the way to run the test suite :)

Most of the failures are harmless, they concern how I changed some defaults. This will be fixed for the 0.0.4 release

**Comment by cournape** [comment]
TST+REF: fix visitor tests + do not enforce defaults in dispatcher.

Enforcing default values for a package should not be done at the AST
level (or even parser at all). So we don't do that anymore - we also fix
the numerous related unit tests. Closed by 1008321ac0af96585bf9e795cc24ec719ec3148e.
